### PR TITLE
Enable device emulation, network throttling, and disable disk cache.

### PR DESCRIPTION
**Pull request by paulirish**
This follows the specs from #33.

### Device
* Nexus 5X screensize, DPR, etc.
* Nexus 5X UA spoofing
* Touch events
* Overlay scrollbars, viewport emulation, etc, etc

### Network
* 150ms latency, 1.6Mbps throughput down, 768Kbps up
* Cache disabled.

This PR depends on #88, so here's a cleaner diff between the two:
https://github.com/paulirish/lighthouse/compare/paulirish:cri-encapsulate...emulation

Also now that all this works, we can load m.flipkart successfully, so I've switched the default to that. Results are already interesting. :)

### CODE REVIEW

File: helpers/emulation.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  scale: 1,
+  deviceScaleFactor: 2.625,
+  fitWindow: false,
+  screenOrientation: {
+    angle: 0,
+    type: 'portraitPrimary'
+  }
+};
+
+const DEVICE_EMULATION_USERAGENT = {
```
Comment (brendankenny) [inline helpers/emulation.js]: is there going to be a good way to keep this in sync with shipping Chrome?

Comment (paulirish) [inline helpers/emulation.js]: I hope so, though devtools project hasnt found a good way, yet. :)

For us, we could do something along the lines of pinging the omahaproxy API every week

File: helpers/emulation.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  driver.sendCommand('Emulation.setTouchEmulationEnabled', {
+    enabled: true,
+    configuration: 'mobile'
+  });
+
+  // from emulation/TouchModel.js
+  /* eslint-disable no-proto */ /* global window, document */
+  const injectedFunction = function() {
+    const touchEvents = ['ontouchstart', 'ontouchend', 'ontouchmove', 'ontouchcancel'];
+    var recepients = [window.__proto__, document.__proto__];
```
Comment (brendankenny) [inline helpers/emulation.js]: wow

File: helpers/emulation.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/**
```
Comment (brendankenny) [inline helpers/emulation.js]: we're going to have to restructure `helpers/` soon

File: index.js [hunk 4]
```diff
@@ -16,7 +16,7 @@
  */
 'use strict';
 
-const defaultUrl = 'https://voice-memos.appspot.com';
+const defaultUrl = 'https://m.flipkart.com';
```
Comment (brendankenny) [inline index.js]: do we want `m.flipkart.com` or allow it to redirect from `flipkart.com` if it wants to?

**Comment by paulirish** [comment]
review?

**Comment by brendankenny** [comment]
you'll have to fix your merge conflicts

**Comment by paulirish** [comment]
fixd

File: helpers/browser/driver.js [hunk 5]
```diff
@@ -265,6 +269,15 @@ class ChromeProtocol {
       });
     });
   }
+
+  beginEmulation() {
+    return new Promise((resolve, reject) => {
+      emulation.enableNexus5X(this);
+      emulation.enableNetworkThrottling(this);
+      emulation.disableCache(this);
+      this.pendingCommandsComplete().then(_ => resolve());
```
Comment (samccone) [inline helpers/browser/driver.js]: :heart:

**Pull request merged**